src/main/java/seedu/address/logic/commands/DeleteTagCommand.java

```diff
@@ -26,123 +23,71 @@ public class DeleteTagCommand extends Command {
-     * @return CommandResult indicating the outcome of the deletion attempt.
-     */
-    private CommandResult createCommandResult(boolean isSuccessful) {
-        if (!isSuccessful) {
-            return new CommandResult(MESSAGE_NONEXISTENT);
-        }
-        return new CommandResult(MESSAGE_SUCCESS);
-    }
-
-    /**
-     * Executes the DeleteTagCommand by deleting the specified tags from the tag list in the model.
-     * If the tags are currently being used to tag any persons, the user is prompted for confirmation.
-     *
-     * @param model Model containing the tags and persons to operate on.
-     * @return CommandResult indicating the result of the command.
-     * @throws CommandException If the user cancels the deletion confirmation.
-     */
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireAllNonNull(model);
 
-        // Get a set of tags currently being used to tag any Person.
-        Set<Tag> tagsInUse = model.getTagsInUse();
-
-        if (!tagsInUse.isEmpty()) {
-            getUserConfirmation(tagsInUse);
+        for (Tag tag : tags) {
+            if (!model.hasTag(tag)) {
+                return new CommandResult(MESSAGE_NONEXISTENT);
+            }
         }
 
-        boolean isSuccessful = model.deleteTags(tags);
-        removeTagsFromPersons(model);
-
-        return createCommandResult(isSuccessful);
-    }
-
-    /**
-     * Prompts the user for confirmation before deleting tags that are currently in use by any persons.
-     *
-     * @param tagsInUse Set of tags that are in use by persons in the model.
-     * @throws CommandException If the user chooses to cancel the deletion.
-     */
-    private void getUserConfirmation(Set<Tag> tagsInUse) throws CommandException {
-        String tagsInUseString = tagsInUse.stream()
-                .map(Tag::toString)
-                .collect(Collectors.joining(", "));
-
-        String confirmationMessage = String.format(MESSAGE_CONFIRMATION, tagsInUseString);
-        boolean isConfirmedDeletion = UserConfirmation.getConfirmation(confirmationMessage);
-        if (!isConfirmedDeletion) {
-            throw new CommandException(MESSAGE_CANCELLED);
+        // Proceed with force deletion logic
+        if (isForceDelete) {
+            for (Tag tag : tags) {
+                model.deleteTags(List.of(tag)); // Remove from model tags
+                model.removeTagFromPersons(tag); // Remove from all persons
+            }
+            return new CommandResult(MESSAGE_SUCCESS_FORCE);
+        } else {
+            // Check if any tags are in use
+            Set<Tag> tagsInUse = model.getTagsInUse();
+            Set<Tag> matchingTags = tagsInUse.stream().filter(tag -> tags.contains(tag)).collect(Collectors.toSet());
+
+            if (!matchingTags.isEmpty()) {
+                return new CommandResult(String.format(MESSAGE_TAGS_IN_USE, tagsInUse));
```

Comment:
Should this be `CommandResult(String.format(MESSAGE_TAGS_IN_USE, matchingTags));` instead? Since we are only concerned with the tags that are being used and in the deletetag command

Reply:
yes boss, issue resolved